### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR:

- Understand the concept of state management.
- Use store, actions, and reducers in React.

### CODE REVIEW

File: src/components/BookSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => state.filter((book) => book.id !== action.payload),
+  },
+});
+
+export const { addBook, removeBook } = booksSlice.actions;
+
+export default booksSlice.reducer;
```
Comment (julie-ify) [inline src/components/BookSlice.js]: - [x] This file should be located here. It should be in the `redux/books` folder and saved using the ducks pattern. It should be in this file.

  ```
  |---/redux
    |---/books
      | bookSlice.js
  ```

File: src/components/CategoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const initialState = 'Under construction';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
+
+export default categoriesSlice.reducer;
```
Comment (julie-ify) [inline src/components/CategoriesSlice.js]: - [x] This file should be located here. It should be in the `redux/categories` folder and saved using the ducks pattern. It should be in this file.

  ```
  |---/redux
    |---/categories
      | categoriesSlice.js
  ```

File: src/components/CategoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under construction';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
```
Comment (julie-ify) [inline src/components/CategoriesSlice.js]: - [ ] Kindly create a `checkStatus` reducer that returns the 'Under construction' text.

File: src/components/CategoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under construction';
```
Comment (julie-ify) [inline src/components/CategoriesSlice.js]: - [x] The initial state should be an empty array. Kindly fix that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mpa-mxiang,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/CategoriesSlice.js (hunk 3 above)
Comment (mpa-mxiang) [reply src/components/CategoriesSlice.js]: Hi, @julie-ify  should I just change it like this: `reducers: { checkStatus }`?

Comment (mpa-mxiang) [reply src/components/CategoriesSlice.js]: Hi @julie-ify, `reducers: { checkStatus }` cannot pass eslint check, do you want me to delete `reducers: {}` and replace it as `checkStatus: 'Under construction'`?

**Review by DammyShittu — APPROVED** [review]
Hi @mpa-mxiang,

## APPROVED 🟢

## Highlights
✔️ Reducers and actions

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mpa-mxiang** [comment]
Thank you for the approval! @DammyShittu

**Pull request merged**